src/core/jit/exec.rs

```diff
@@ -171,8 +184,41 @@ where
     }
 }
 
+#[derive(Clone)]
+pub struct ExecResult<V> {
+    pub value: V,
+    pub type_name: Option<TypeName>,
+}
+
+pub struct ExecResultRef<'a, V> {
+    pub value: &'a V,
+    pub type_name: Option<&'a TypeName>,
+}
```

Comment:
This seems like is morally equivalent to a `Cow<'a, TypedValue<&'a V>>`.

Reply:
no, `type_name` is also a reference here while in base struct its an owned value